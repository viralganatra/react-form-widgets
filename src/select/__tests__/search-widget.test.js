import React from 'react';
import SearchWidget from '../search-widget';

jest.mock('../../input/single/input', () => jest.fn(
    ({ inputRef, ...rest }) => <input {...rest} ref={inputRef} />,
));

describe('SelectSearchWidget', () => {
    it('should render correctly, setting the focus on mount', () => {
        const wrapper = mount(<SearchWidget />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('input').node).toBe(document.activeElement);
    });
});
