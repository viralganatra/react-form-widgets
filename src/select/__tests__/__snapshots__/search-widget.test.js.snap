exports[`SelectSearchWidget should render correctly, setting the focus on mount 1`] = `
<SelectSearchWidget>
  <mockConstructor
    inputRef={[Function]}>
    <input />
  </mockConstructor>
</SelectSearchWidget>
`;
